client/reader/discover/discover-stream.js

```diff
@@ -1,131 +1,36 @@
 import { getReaderFollowedTags } from 'calypso/state/reader/tags/selectors';
 import {
 	getDiscoverStreamTags,
 	DEFAULT_TAB,
-	getSelectedTabTitle,
 	buildDiscoverStreamKey,
-	FIRST_POSTS_TAB,
 	ADD_NEW_TAB,
 	REDDIT_TAB,
 } from './helper';
-import './style.scss';
-
-const DISCOVER_HEADER_NAVIGATION_ITEMS = [];
-
-export const DiscoverHeader = ( props ) => {
-	const translate = useTranslate();
-
-	const { selectedTab } = props;
-	const tabTitle = getSelectedTabTitle( selectedTab );
-
-	let subHeaderText;
-	switch ( selectedTab ) {
-		case FIRST_POSTS_TAB:
-			subHeaderText = translate(
-				'Fresh voices, fresh views. Explore first-time posts from new bloggers.'
-			);
-			break;
-		case ADD_NEW_TAB:
-			subHeaderText = translate( 'Subscribe to new blogs, newsletters, and RSS feeds.' );
-			break;
-		case REDDIT_TAB:
-			subHeaderText = translate( 'Follow your favorite subreddits inside the Reader.' );
-			break;
-		default:
-			subHeaderText = translate( 'Explore %s blogs that inspire, educate, and entertain.', {
-				args: [ tabTitle ],
-				comment: '%s is the type of blog being explored e.g. food, art, technology etc.',
-			} );
-	}
-
-	return (
-		<NavigationHeader
-			navigationItems={ DISCOVER_HEADER_NAVIGATION_ITEMS }
-			title={ translate( 'Discover' ) }
-			subtitle={ subHeaderText }
-			className={ clsx( 'discover-stream-header', {
-				'reader-dual-column': props.width > WIDE_DISPLAY_CUTOFF,
-			} ) }
-		/>
-	);
-};
 
 const DiscoverStream = ( props ) => {
 	const translate = useTranslate();
-	const currentLocale = useLocale();
 	const followedTags = useSelector( getReaderFollowedTags );
 	const isLoggedIn = useSelector( isUserLoggedIn );
 	const selectedTab = props.selectedTab || DEFAULT_TAB;
-	const selectedTag = props.query?.selectedTag;
-
-	// If the selected tab is tags and no selectedTag is provided, redirect to the tags tab with dailyprompt selected.
-	if ( selectedTab === 'tags' && ! selectedTag ) {
-		const redirectPath = '/discover/tags';
-		const localizedPath = addLocaleToPathLocaleInFront( redirectPath, currentLocale );
-		return page.redirect( addQueryArgs( { selectedTag: 'dailyprompt' }, localizedPath ) );
-	}
-
-	const isDefaultTab = selectedTab === DEFAULT_TAB;
-
-	// Do not supply a fallback empty array as null is good data for getDiscoverStreamTags
-	const recommendedStreamTags = getDiscoverStreamTags(
-		followedTags && followedTags.map( ( tag ) => tag.slug ),
-		isLoggedIn
-	);
+	const selectedTag = props.query?.selectedTag ?? 'dailyprompt';
 
 	const effectiveTabSelection = 'tags' === selectedTab ? selectedTag : selectedTab;
-	const streamKey = buildDiscoverStreamKey( effectiveTabSelection, recommendedStreamTags );
-
-	const handleTagSelect = ( tag ) => {
-		const redirectPath = '/discover/tags';
-		const localizedPath = addLocaleToPathLocaleInFront( redirectPath, currentLocale );
-		page.replace( addQueryArgs( { selectedTag: tag }, localizedPath ) );
-	};
-
-	const streamProps = {
```

Comment:
Using props directly in `<Stream>` component instead of defining it as a separate variable.